File: paypal/692. Top K Frequent Words.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        arr = defaultdict(int)
        for i in words:
            arr[i]+=1
        
        sorted_items = sorted(arr.items(), key=lambda x: (-x[1], x[0]))
        
        return [word for word, _ in sorted_items[:k]]
        

File: java_codes/692. Top K Frequent Words.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        
        HashMap<String,Integer> map = new HashMap<>();
        for(String s:words)
        {
            map.put(s,map.getOrDefault(s,0)+1);
        }


        List<String> list = new ArrayList(map.keySet());

        Collections.sort(list, (i, j) -> map.get(i).equals(map.get(j)) ? i.compareTo(j) : map.get(j) - map.get(i));


        return list.subList(0,k);

    }
}